File: backend/src/main/java/com/ums/service/AuthService.java
package com.ums.service;

import com.ums.dto.auth.LoginRequest;
import com.ums.dto.auth.LoginResponse;
import com.ums.dto.auth.TokenResponse;
import com.ums.dto.auth.RefreshRequest;

public interface AuthService {
    
    LoginResponse login(String usernameOrEmail, String password, String ipAddress, String userAgent);
    
    TokenResponse refreshToken(RefreshRequest request);
    
    void logout(String token);
    
    boolean verifyMfa(String code, String tempToken);
}

File: frontend/src/services/authService.ts
import { apiService } from './api';

export const authService = {
  login: async (credentials: { usernameOrEmail: string; password: string }) => {
    // Tokens are set via httpOnly cookies by the backend
    const response = await apiService.clientInstance.post('/auth/login', credentials);
    return response;
  },

  refreshToken: async () => {
    // Refresh token is read from cookie by backend
    return apiService.clientInstance.post('/auth/refresh', {});
  },

  logout: async () => {
    return apiService.clientInstance.post('/auth/logout');
  },

  getCurrentUser: async () => {
    // Cookies are automatically sent with requests (withCredentials: true)
    return apiService.clientInstance.get('/users/me');
  },

  verifyMfa: async (code: string, tempToken: string) => {
    return apiService.clientInstance.post('/auth/mfa/verify', { code }, {
      headers: { 'X-Auth-Token': tempToken }
    });
  },
};
